src/selectors/entities/users.ts

```diff
@@ -344,6 +353,17 @@ export function searchProfiles(state: GlobalState, term: string, skipCurrent = f
     return filteredProfiles;
 }
 
+export function searchProfilesInChannel(state: GlobalState, channelId: $ID<Channel>, term: string, skipCurrent = false): Array<UserProfile> {
+    const doGetProfilesInChannel = makeGetProfilesInChannel();
+    const profiles = filterProfilesMatchingTerm(doGetProfilesInChannel(state, channelId, false), term);
+
+    if (skipCurrent) {
+        removeCurrentUserFromList(profiles, getCurrentUserId(state));
```

Comment:
Out of curiosity, we don't need to reassign profiles here? It will change profiles on the spot without need to re-assign? I just thought javascript was always pass by value and pass by reference wasn't a thing although I may be incorrect.

Reply:
I noticed we use this elsewhere but I'm still confused how removeCurrentUserFromList effects `profiles` when Javascript is a pass by value language. Do you happen to know how it works cause I don't quite understand it although we are doing the same exact thing in other parts of the codebase.